### PR TITLE
feat: add a method to reclaim free space on a cluster

### PR DESCRIPTION
**Pull request by gssbzn**
## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: atmcfg/atmcfg.go [hunk 1]
```diff
@@ -419,3 +419,51 @@ func Restart(out *opsmngr.AutomationConfig, name string) {
...
+// ReclaimFreeSpace sets all process of a cluster to reclaim free space
+func ReclaimFreeSpace(out *opsmngr.AutomationConfig, name string) {
+	// This value may not be present and is mandatory
+	if out.Auth.DeploymentAuthMechanisms == nil {
+		out.Auth.DeploymentAuthMechanisms = make([]string, 0)
+	}
+	lastCompact := time.Now().Format(time.RFC3339)
+	reclaimByReplicaSetName(out, name, lastCompact)
+	reclaimByShardName(out, name, lastCompact)
+}
```
Comment (andreaangiolillo) [inline atmcfg/atmcfg.go:432]: [q] Where are we going to use this function in mongocli?

**Review by andreaangiolillo — APPROVED** [review]
LGTM.

File: atmcfg/atmcfg.go (hunk 1 above)
Comment (gssbzn) [reply atmcfg/atmcfg.go:432]: https://feedback.mongodb.com/forums/930808-mongodb-cli/suggestions/41674054-allow-to-perform-rolling-compact-via-restapi

Comment (gssbzn) [reply atmcfg/atmcfg.go:432]: Also I'm trying to confirm if it makes sense to set this for mongos as I think maybe not

**Pull request merged**